docs/libraries/typespec-client-generator-core/reference/decorators.md

```diff
@@ -172,6 +172,34 @@ model ModelToExclude {
 }
 ```
 
+### `@flattenProperty` {#@Azure.ClientGenerator.Core.flattenProperty}
+
+Set whether a model property should be flattened or not.
+
+```typespec
```

Comment:
Do we want to mention in the doc that this decorator is not recommended to use?

Reply:
This doc is auto-generated. I actually have `#deprecated "@flattenProperty decorator is not recommended to use."` in the original code comment.